### ISSUE TITLE
service-client-sample using wrong Java compiler version 1.6

### ISSUE THREAD

**Issue by JMayrbaeurl** [Azure/azure-iot-sdk-java#309]
- **OS and version used:** Windows 10

- **Java runtime used:** 1.8

- **SDK version used:** Service 1.14.1

# Description of the issue:
Java 8 class CompleteableFuture used in code, but compiler settings set to Java 1.6

# Code sample exhibiting the issue:
In main method of samples.com.microsoft.azure.sdk.iot.ServiceClientSample

```
        // Receive FileUploadNotification
        CompletableFuture<FileUploadNotification> fileUploadNotificationCompletableFuture = fileUploadNotificationReceiver.receiveAsync(RECEIVER_TIMEOUT);
        FileUploadNotification fileUploadNotification = fileUploadNotificationCompletableFuture.get();

```
and pom.xml

```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.3</version>
                <configuration>
                    <source>1.6</source>
                    <target>1.6</target>
                </configuration>
            </plugin>

```

# Need Support?
* Have a feature request for SDKs? Please post it on [User Voice](https://feedback.azure.com/forums/321918-azure-iot) to help us prioritize.
* Have a technical question? Ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-iot-hub) with tag “azure-iot-hub”
* Need Support? Every customer with an active Azure subscription has access to support with guaranteed response time.  Consider submitting a ticket and get assistance from Microsoft support team
* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C, Java, .NET, Node.js, Python).

**Comment by jasmineymlo** [comment]
fixed in 4ff84cfe90358ba0c759a6fa606cca03878254fb

**Comment by timtay-microsoft** [comment]
The fix for this issue has been released as of Service Client 1.15.0, so I'm closing this issue now.

**Comment by az-iot-builder-01** [comment]
@JMayrbaeurl, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
